test/sidecar_test.go

```diff
@@ -47,26 +47,32 @@ func TestSidecarTaskSupport(t *testing.T) {
 		stepCommand    []string
 		sidecarCommand []string
 	}{{
-		desc:           "A sidecar that runs forever is terminated when Steps complete",
-		stepCommand:    []string{"echo", "\"hello world\""},
-		sidecarCommand: []string{"sh", "-c", "while [[ true ]] ; do echo \"hello from sidecar\" ; done"},
+		desc:        "A sidecar that runs forever is terminated when Steps complete",
+		stepCommand: []string{"echo", "\"hello world\""},
+		// trapping the exit signals lets us stop the sidecar more
+		// rapidly than waiting for a force stop.
+		sidecarCommand: []string{"sh", "-c", "trap exit SIGTERM SIGINT; while [[ true ]] ; do echo \"hello from sidecar\" ; done"},
 	}, {
 		desc:           "A sidecar that terminates early does not cause problems running Steps",
 		stepCommand:    []string{"echo", "\"hello world\""},
 		sidecarCommand: []string{"echo", "\"hello from sidecar\""},
 	}}
 
 	ctx := context.Background()
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	clients, namespace := setup(ctx, t)
 	t.Parallel()
 
-	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, clients, namespace) }, t.Logf)
-	defer tearDown(ctx, t, clients, namespace)
-
 	for i, test := range tests {
+		i := i
+		test := test
 		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
```

Comment:
do we still want to call t.Parallel above as well?

Reply:
I wasn't entirely sure how t.Parallel in sub-tests interacted with other parallel tests but after reading the docs on [Parallel](https://pkg.go.dev/testing#T.Parallel) and [Run](https://pkg.go.dev/testing#T.Run) it _seems_ like all parallel tests are run together regardless whether they're sub-tests or top-level tests.  removed the outer t.Parallel() 👍

Reply:
@sbwsg interesting. Because the parent test is not `t.Parallel` it won't be executed in parallel of the other though right ? Making those subtests be run in parallel of each other but not in parallel with "external" tests. Not sure if that's the expected behavior your are looking for or not ?

Reply:
> Because the parent test is not t.Parallel it won't be executed in parallel of the other though right ?

This is the part i wasn't sure about. Here's the docs for `t.Run()`:

> Run runs f as a subtest of t called name. It runs f in a separate goroutine and blocks until f returns or calls t.Parallel to become a parallel test.

it doesn't describe the behaviour of parent / child tests running in parallel. So I went back and read the rest of the testing package docs and found this quote from the intro:

> Subtests can also be used to control parallelism. A parent test will only complete once all of its subtests complete.

I _think_ this matches what you said but I am still having a hard time parsing the docs.  So I put back the `t.Parallel` in the parent test.  I want everything (external tests + sidecar subtests) to run all together.